code/modules/materials/Mat_RawMaterials.dm

```diff
@@ -43,12 +44,35 @@
 		P.change_stack_amount(toRemove - P.amount)
 		return P
 
-	attackby(var/obj/item/W as obj, mob/user as mob)
-		if(istype(W, /obj/item/material_piece) && W.material)
-
-			if(src.stack_item(W))
-				boutput(user, "<span class='notice'>You stack \the [W]!</span>")
-		return
+	attack_hand(mob/user as mob)
+		if((user.r_hand == src || user.l_hand == src) && src.amount > 1)
+			var/splitnum = round(input("How many material pieces do you want to take from the stack?","Stack of [src.amount]",1) as num)
+			var/diff = src.amount - splitnum
+			if (splitnum >= amount || splitnum < 1)
+				boutput(user, "<span class='alert'>Invalid entry, try again.</span>")
+				return
+			boutput(user, "<span class='notice'>You take [splitnum] material pieces from the stack, leaving [diff] pieces behind.</span>")
+			src.amount = diff
+			var/obj/item/material_piece/new_stack = new src.type(user.loc, diff)
+			new_stack.amount = splitnum
+			new_stack.attack_hand(user)
+			new_stack.add_fingerprint(user)
+			new_stack.update_stack_appearance()
```

Comment:
not a fan of how this code is duplicated

Reply:
right, I managed to refactor the code, it's a bit less dumb now